### DISCUSSION TITLE
Incorrect Confirmation Dialog Presentation on iPad

### DISCUSSION THREAD

**Discussion by bradhowes** [pointfreeco/swift-composable-architecture#3130] [General]
### Description

In the "Case Studies > Alerts & Dialogs" form, clicking on "Confirmation Dialog" should show a confirmation sheet when run on iPad, but it instead shows a  truncated popup:

<img width="732" alt="Capto_Capture 2024-05-31_11-23-01_PM2" src="https://github.com/pointfreeco/swift-composable-architecture/assets/686946/4bec0d3f-32f5-4e99-ab19-5cf5929faddb">

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Similar/same behavior should be encountered on iPad as seen on iPhone.

### Actual behavior

See picture above

### Steps to reproduce

1. Run the Case Studies application on iPad
2. Visit the "Alerts & Dialogs"
3. Tap on the "Confirmation Dialog" button

### The Composable Architecture version information

fb5159e

### Destination operating system

iOS 17.5

### Xcode version information

Version 15.4 (15F31d)

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @bradhowes, I may be misunderstanding what you are describing, but the screenshot you shared _is_ a confirmation dialog on iPad. They are rendered as popovers on iPad. 

Can you explain a bit more about what you think is wrong in the screenshot? Have you tried creating a confirmation dialog in vanilla SwiftUI to confirm what it looks like?

Since I don't believe this is an issue with the library I am going to convert it to a discussion. Feel free to continue the conversation over there.

**Comment by bradhowes** [comment]
@mbrandonw IMO the behavior is confusing -- there are two buttons in the confirmation dialog definition, but only one is shown. However, if you attach the ".confirmationDialog(...)" to the "Confirmation Dialog" button instead of the form, you get what I consider proper behavior.

<img width="503" alt="Capto_Capture 2024-06-01_10-29-51_PM2" src="https://github.com/pointfreeco/swift-composable-architecture/assets/686946/6af4d6e0-27da-4d31-874a-6c726b459b5e">
